src/components/App/index.js

```diff
@@ -30,12 +58,12 @@ export class App extends Component {
            <Switch>
              <Route exact path='/' component={StartScreen} />
              <Route exact path='/rules' component={RuleList}/>
-             <Route exact path='/new' component={NewGame} />
+             <Route exact path='/new' render={()=> <NewGame handleUserInit={this.handleUserInit}/>} />
              <Route exact path='/join' component={JoinGame} />
              <Route exact path='/lobby' component={Lobby} />
              <Route exact path='/game' component={Main} />
              <Route component={ErrorScreen} />
-             <Route path='/' render={() => <Main websocket={this.state.socket}/>}/>
+             <Route path='/' render={() => <Main token={this.state.token}/>}/>
```

Comment:
So we pass the token that was generated by the user input over in NewGame?
Then Main uses the token to handle the "conversational component" ??

Reply:
Yes, thats how it knows the channel URL